app/models/user.rb

```diff
@@ -9,4 +9,47 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  has_many :friendships
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
+  has_many :friends, through: :friendships, foreign_key: 'friend_id'
+
+  def friends
+    friends_array = friendships.map { |friendship| friendship.friend if friendship.confirmed }
+    inverse_friends_array = inverse_friendships.map { |friendship| friendship.user if friendship.confirmed }
+    (friends_array + inverse_friends_array).compact
+  end
+
+  def pending_friends
+    friendships.map { |friendship| friendship.friend unless friendship.confirmed }.compact
+  end
```

Comment:
- [ ] User#pending_friends can be defined as association with SQL query instead of processing objects in application memory (as the database was created to store and query data it is always more efficient than ruby). Rewrite it to use SQL query. Now you understand better why we prefer SQL query:blush:.